### PR TITLE
Use dynamic template for test results email

### PR DESCRIPTION
**Pull request by zdeveloper**
## Related Issue or Background Info

- part 3 of #1950 using dynamic template

## Changes Proposed

- use dynamic template to send test result email

### UI
- [x] Any changes to the UI/UX are approved by design 
- [ x] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [x] Database changes are submitted as a separate PR
  - [x] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [x] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [x] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [x] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [x] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [x] Any dependencies introduced have been vetted and discussed

## Cloud
- [ ] DevOps team has been notified if PR requires ops support
- [ ] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/service/email/EmailService.java [hunk 1]
```diff
@@ -133,4 +133,23 @@ public String sendWithProviderTemplate(
 
     return emailProvider.send(mail);
   }
+
+  public String sendWithDynamicTemplate(
```
Comment (bmaden-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/email/EmailService.java:137]: Not a blocker, but might consider making the name of this method match the other one for sending with a dynamic template without variables (or maybe the other way around?... this name is probably more clear).

File: backend/src/test/java/gov/cdc/usds/simplereport/service/TestResultsDeliveryServiceTest.java [hunk 2]
```diff
@@ -57,22 +50,13 @@ void emailTestResultTests() throws IOException {
...
-    assertThat(emailTemplate.getTestResultUrl())
-        .isEqualTo("https://simplereport.gov/pxp?plid=" + uuid);
-    assertThat(emailTemplate.getExpirationDuration()).isEqualTo("2 days");
+        .sendWithDynamicTemplate(
+            "harry@hogwarts.edu",
+            EmailProviderTemplate.SIMPLE_REPORT_TEST_RESULT,
+            Map.of(
+                "facility_name", "House of Gryffindor",
+                "expiration_duration", "2 days",
+                "test_result_url", "https://simplereport.gov/pxp?plid=" + uuid));
```
Comment (bmaden-usds) [inline backend/src/test/java/gov/cdc/usds/simplereport/service/TestResultsDeliveryServiceTest.java:59]: Nice

**Review by bmaden-usds — APPROVED** [review]
Thanks for doing this!

Is the Waitlist email the only one still using the old template provider stuff?  Maybe we can add a ticket to strip out the rest of it.  We don't even use the Waitlist, so if that's all that's left then we should be able to go ahead and clean it up.

File: backend/src/main/java/gov/cdc/usds/simplereport/service/email/EmailService.java (hunk 1 above)
Comment (zdeveloper) [reply backend/src/main/java/gov/cdc/usds/simplereport/service/email/EmailService.java:137]: sure thing, I'll swab the other method to this name

**Comment by zdeveloper** [comment]
@bmaden-usds It seems like we have 2 emails that are still using the html template. I made this [ticket #2865 ](https://github.com/CDCgov/prime-simplereport/issues/2865) to clean them up

**Pull request merged**